datasets.py

```diff
@@ -73,3 +73,15 @@ def from_text(cls, text: str) -> Self:
             dp = DataPoint(row_values[0], row_values[1], row_values[2])
             data.append(dp)
         return cls(type_, accepted, reference, ionic_liquid, details, data)
+
+    @classmethod
+    def from_file(cls, file_path: str) -> list[Self]:
+        """Convert a file containing multiple datasets into a list of `Dataset`
+        instances.
+        """
+        with open(file_path, "r") as file:
+            file_data = file.read().split("\n\n")
+        datasets_list = cast(
```

Comment:
I think you don't need to apply casting here. You can refer to the class simply by `cls` and then `mypy` will not complain...

Try:

```python
[cls.from_text(text) for text in file_data]
```